packages/core/src/app/checkout/AnalyticsEvents.ts

```diff
@@ -21,34 +19,79 @@ export enum GuestCheckoutEvents {
+// borrowed polling mechanism 
+async function pollUntilAccepted<T>(
+  fn: () => T,
+  accept: (t: T) => boolean,
+  timeoutMillis = defaultTimeout,
+  interval = defaultPollingInterval,
+): Promise<T | Error> {
+  return new Promise<T | Error>(resolve => {
+    let remainingTime = timeoutMillis;
+
+    const intervalId = setInterval(() => {
+      if (remainingTime < 0) {
+        clearInterval(intervalId);
+        resolve(new Error(`Timed out while waiting for ${fn} (${timeoutMillis}ms)`));
+        return;
+      }
+
+      const value = fn();
+      if (accept(value)) {
+        const timeElapsed = timeoutMillis - remainingTime;
+        console.log(`--bolt bigc--: poll condition accepted after ${timeElapsed} ms`);
+        clearInterval(intervalId);
+        resolve(value);
+        return;
+      }
+
+      remainingTime = remainingTime - interval;
+    }, interval);
+  });
+}
+
+function checkForBoltAnalytics() {
+  try {
+    if (window && window.BoltAnalytics) {
+      boltTracker = window.BoltAnalytics;
+      console.log("--bolt bigc--: successfully found window.BoltAnalytics");
+      return true;
+    }
+  } catch (e) {
+    console.log("--bolt bigc--: error while initializing AnalyticsEvents", e);
+  }
+  return false;
+}
+
 export const AnalyticsEvents = {
   init: () => {
     try {
-      console.log("--bolt bigc--: intializing Bolt-BigC analytics events");
-      if (window && window.BoltTrack) {
-        boltTracker = window.BoltTrack;
-        console.log("--bolt bigc--: successfully assigned window BoltTrack");
-        // immediately send checkout load success event
-        AnalyticsEvents.emitEvent("Checkout load success");
-      }
+      // immediately send checkout load success event
+      AnalyticsEvents.emitEvent("Checkout load success");
+      // poll until window.BoltAnalytics is on the page
+      pollUntilAccepted(checkForBoltAnalytics, t => t, 3000)
+        .catch(e => console.warn("--bolt bigc--: failure to locate BoltAnalytics on the window", e));
     } catch (e) {
-      console.log("--bolt bigc--: error during emitting event ", e);
+      console.log("--bolt bigc--: error during polling for BoltAnalytics", e);
     }
   },
   emitEvent: (eventName: string, eventProps?: any) => {
     try {
-      console.log("--bolt bigc--: emitting analytics event ", eventName);
-      const props: any = {
-        ...eventProps,
-        nextState: eventName,
-        prevState: "",
-      };
-      if (eventLog.length > 0) {
-        props.prevState = eventLog[0];
+      eventQueue.push({ eventName, eventProps });
+      // wait until checkoutStepComplete is defined to emit logs
+      if (!boltTracker || !boltTracker.checkoutStepComplete) {
+        return;
+      }
+
+      while (eventQueue.length) {
+        const { eventName, eventProps } = eventQueue.shift();
+
+        console.log("--bolt bigc--: emitting analytics event ", eventName);
+
+        boltTracker.checkoutStepComplete(eventName, eventProps);
```

Comment:
What happens if this fails? Are we treating this as best-efforts? (i think we should given its not critical)